keps/sig-network/2091-cluster-network-policy/README.md

```diff
@@ -0,0 +1,999 @@
+the cluster, all Pods in that Namespace should only allow traffic from Pods in
+the _same Namespace_ who has label app=b. Hence, the policy above allows
+x/b1 -> x/a1 and y/b2 -> y/a2, but denies y/b2 -> x/a1 and x/b1 -> y/a2.
+
+#### IPBlock
+
+The `ClusterNetworkPolicyPeer` and `ClusterDefaultNetworkPolicyPeer` both allow the
+ability to set an `IPBlock` as a peer. The usage of this field is similar to
+how it is used in the NetworkPolicyPeer. However, we should also explicitly
+note that the IPBlock set in this field could belong to the cluster locally, or
+could be cluster external. For example, the peer could be set with a subnet
+or an IP which maps to a Pod existing in the cluster. This means that a Pod
+in a cluster-scoped NetworkPolicy could be identified by either the labels
+applied on the Pod, or its IP address. In case of multiple conflicting rules
+targeting the same Pod, but identified in different ways, such as via labelSelector
+or via PodIP set in IPBlock, the net effect of the rules will be determined
+by the `action` associated with the rule and/or the resource in which the
+rule is set in, i.e. ClusterNetworkPolicy rule takes precedence over a
+ClusterDefaultNetworkPolicy rule.
+
+### Sample Specs for User Stories
+
+![Alt text](user_story_diagram.png?raw=true "User Story Diagram")
+
+#### Story 1: Deny traffic from certain sources
+As a cluster admin, I want to explicitly deny traffic from certain source IPs
+that I know to be bad.
+
+```yaml
+apiVersion: networking.k8s.io/v1alpha1
+kind: ClusterNetworkPolicy
+metadata:
+  name: deny-bad-ip
+spec:
+  appliedTo:
+    # if there's an ingress gateway in the cluster, applying the policy to
+    # gateway namespace will be sufficient
+    namespaceSelector: {}
+  ingress:
+    - action: Deny
+      from:
+      - ipBlock:
+          cidr: 192.0.2.0/24  # banned addresses
+```
+
+#### Story 2: Funnel traffic through ingress/egress gateways
+As a cluster admin, I want to ensure that all traffic coming into (going out of)
+my cluster always goes through my ingress (egress) gateway.
+
+```yaml
+apiVersion: networking.k8s.io/v1alpha1
+kind: ClusterNetworkPolicy
+metadata:
+  name: ingress-egress-gateway
+spec:
+  appliedTo:
+    namespaceSelector:
+      matchLabels:
+        type: tenant  # assuming all tenant namespaces will be created with this label
+  ingress:
+    - action: Authorize
+      from:
+      - namespaces:
+          selector:
+            matchLabels:
+              kubernetes.io/metadata.name: dmz # ingress gateway
+    - action: Deny
+      from:
+      - ipBlock:
+          cidr: 0.0.0.0/0
+  egress:
+    - action: Authorize
+      to:
+      - namespaces:
+          selector:
+            matchLabels:
+              kubernetes.io/metadata.name: istio-egress  # egress gateway
+    - action: Deny
+      to:
+      - ipBlock:
+          cidr: 0.0.0.0/0
+```
+
+__Note:__ The above policy is very restrictive, i.e. it rejects ingress/egress
```

Comment:
> I don't think we can use such a model as an example - it's not a great solution to the problem presented :)

Because it is too restrictive to start with?